python/tk_multi_publish2/api/plugins/publish_plugin_instance.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2018 Shotgun Software Inc.
```

Comment:
Basically same as this: https://github.com/shotgunsoftware/tk-multi-publish2/blob/master/python/tk_multi_publish2/processing/plugin.py#L118 with the added get/set state methods used for pickling.